ui/src/hooks/user.tsx

```diff
@@ -53,41 +39,81 @@ export function useLoginWithToken() {
-  return useQuery<UserInterface>({
-    queryKey: ['getUser'],
-    queryFn: async () => {
-      return apiClient
-        .get(`/scim2/Users/self`, {
-          baseURL: `${UC_AUTH_API_PREFIX}`,
-        })
-        .then((response) => response.data)
-        .catch((error) => {
-          if (error?.status === HttpStatus.UNAUTHORIZED) {
-            return null;
-          } else {
-            throw new Error('Failed to fetch user');
-          }
-        });
-    },
-  });
+// TODO:
+// As of [28/11/2024], the OpenAPI specification for auth-related APIs has not been defined.
+// Once the specification is added, the following hooks should be updated.
+// SEE:
+// https://github.com/unitycatalog/unitycatalog/issues/768
+enum HttpStatus {
+  OK = 200,
+  CREATED = 201,
+  BAD_REQUEST = 400,
+  UNAUTHORIZED = 401,
+  NOT_FOUND = 404,
+  INTERNAL_SERVER_ERROR = 500,
+}
+
+interface LogoutResponse {
+  response: HttpStatus;
 }
 
 export function useLogoutCurrentUser() {
   return useMutation<LogoutResponse, Error, {}>({
     mutationFn: async () => {
-      return apiClient
-        .post(
-          `/auth/logout`,
-          {},
-          {
-            baseURL: `${UC_AUTH_API_PREFIX}`,
-          },
-        )
+      return CLIENT.post(`/auth/logout`, {})
         .then((response) => response.data)
         .catch((e) => {
           throw new Error(e.response?.data?.message || 'Logout method failed');
         });
     },
   });
 }
+
+export type UserInterface = ApiInterface<ControlComponent, 'UserResource'>;
+
+export function useGetCurrentUser() {
+  const expectedErrorCodes = [
+    401, // UNAUTHORIZED
+    404, // NOT_FOUND
+  ] as const;
+
+  type ErrorCode = (typeof expectedErrorCodes)[number];
+
+  const isExpectedError = (response: {
+    status: number;
+    data: any;
+  }): response is ApiErrorResponse<
+    ControlApi,
+    '/scim2/Users/self',
+    'get',
+    ErrorCode
+  > => expectedErrorCodes.map(Number).includes(response.status);
+
+  return useQuery<ApiSuccessResponse<ControlApi, '/scim2/Users/self', 'get'>>({
+    queryKey: ['getUser'],
+    queryFn: async () => {
+      const api = route(
+        CLIENT,
+        {
+          path: '/scim2/Users/self',
+          method: 'get',
+        },
+        isExpectedError,
+      );
+      const response = await api.call();
+      if (response.result !== 'success') {
+        // NOTE:
+        // To the best of my knowledge, most HTTP error codes are not explicitly defined in the OpenAPI
+        // specification. I reviewed the server implementation and identified the error codes related to
+        // authentication by examining the codebase. Relevant details can be found here:
+        // - https://github.com/search?q=repo%3Aunitycatalog%2Funitycatalog%20UNAUTHENTICATED&type=code
+        // - https://github.com/search?q=repo%3Aunitycatalog%2Funitycatalog+NOT_FOUND&type=code
+        if (response.data.status === 401 || response.data.status === 404) {
```

Comment:
I think this should just be checking for 401 status, the 404 was only returning when the endpoint URL was incorrect.